php-frontend/src/main/java/org/sonar/php/PHPAnalyzer.java

```diff
@@ -84,6 +84,12 @@ public void nextFile(PhpFile file) {
     return issuesBuilder.build();
   }
 
+  public void terminate() {
+    for (PHPCheck check : checks) {
+      check.terminate();
```

Comment:
When there's an exception in `analyseFile(...)` a try/catch mechanism log an error and the analysis does not fail. Here, an exception in `check.terminate();` would fail the analysis, I would prefer to have a surrounding try/catch to only log an error.